src/Eleventy.js

```diff
@@ -391,39 +392,68 @@ Arguments:
    * @method
    * @param {String} path - Watch this file.
    */
-  async _watch(path) {
+  async _watch(path, immediate = false) {
     if (path) {
       path = TemplatePath.addLeadingDotSlash(path);
+      this.queuedPaths.push(path);
+      console.log(`Watching ${path}...`);
+      this.lastTriggerTime = new Date();
     }
 
-    if (this.active) {
-      this.queuedToRun = path;
+    if (this.active || this.queuedPaths.length == 0) {
       return;
     }
 
     this.active = true;
 
-    let isInclude =
-      path &&
-      TemplatePath.startsWithSubPath(path, this.eleventyFiles.getIncludesDir());
+    if (this.config.watchTriggerDelay && !immediate) {
+      let timeSinceLastTrigger = new Date() - this.lastTriggerTime;
+
+      while (timeSinceLastTrigger < this.config.watchTriggerDelay) {
+        let triggerDelaySleep =
+          this.config.watchTriggerDelay - timeSinceLastTrigger;
+        console.log(`Waiting ${triggerDelaySleep} ms before processing.`);
+        await sleep(triggerDelaySleep);
```

Comment:
Is it possible to do this with a `setTimeout` instead? Might save a dependency.

Reply:
Yup, I can totally do that. I made a comment about this on the original PR since I think the pulled in library is literally 5 lines of code so it seemed like a bit of overkill but wasn’t sure what your preferences would be.

Reply:
Yup, I can totally do that. I made a comment about this on the original PR since I think the pulled in library is literally 5 lines of code so it seemed like a bit of overkill but wasn’t sure what your preferences would be.